### DISCUSSION TITLE
Bevy Monthly Newsletter Content Submission Policy

### DISCUSSION THREAD

**Discussion by cart** [bevyengine/bevy#4560] [Polls]
Currently Bevy's `#showcase` channel on Discord has the following description:

> Share your Bevy work here! Images, videos, and links are ok. Use threads to discuss posts. Things posted here may be included in our curated This Month in Bevy newsletter. If you don't want us to consider your post for inclusion, please let us know. Feel free to provide relevant links or social media handles that you want us to include in the newsletter.

The question is: how does the community feel about this?

Some alternatives worth considering:
* Require going through a "submit a pull request with your content" model used by This Month in Rust Gamedev. This process will be supported no matter what, so the only question is whether it should be the _only_ process.
* A separate `#newsletter` showcase channel 
* Require #showcase submissions go through a bot that asks for newsletter consent explicitly

The biggest pros of "consent by default in #showcase" are:
* A low barrier to entry for newsletter content submission
* More content in Bevy's monthly newsletter (because `#showcase` will almost certainly be higher traffic than a manual PR process). #showcase is the most active place for visual Bevy content by a pretty wide margin.

And the biggest con is:
* The bar is so low (and so implicit) that people might accidentally consent to including content they don't want to. Many people don't read Discord channel descriptions.

**Comment by aevyrie** [comment]
> The bar is so low (and so implicit) that people might accidentally consent to including content they don't want to. Many people don't read Discord channel descriptions.

I agree this is a very plausible risk. If we end up going the route of implicit consent, I'd like to suggest we ping everyone whose content has been taken from showcase, so it's still opt-in, but we don't add any friction to the actual submission process. Alternatively, this ping could be opt-out instead of opt-in.

**Reply by lukors** [reply]
If we ask people's permission, that means that there no longer is implicit consent, which I think is good. On top of this, we can add in the channel description that people can give consent in their showcase message if they want.

This approach has some nice properties:

- Zero friction when submitting stuff, and if we want to include a post in the newsletter we can just ask permission.
- No risk of "accidentally" giving consent.
- Easily give explicit permission, just give it in the message or thread of the message.

This could be streamlined further if we have a bot that asks people's permission for each message they post on the channel. Don't know how much can be done with a Discord bot, but this doesn't feel too complicated.

**Reply by cart** [reply]
I think manual opt-in in `#showcase` risks being too hidden. If you buy the "its too implicit" argument for auto-consent, then I think that also applies here for giving consent. I think I prefer a second `#newsletter` channel to bump visibility of the feed / process.

**Reply by kurtkuehnert** [reply]
What about a dedicated newsletter emoji showcase authors could put in their message to keep the channels the same? This would only add minimal overhead.

**Reply by HackerFoo** [reply]
A separate channel is easier to discover.

**Reply by lukors** [reply]
> I think manual opt-in in `#showcase` risks being too hidden. If you buy the "its too implicit" argument for auto-consent, then I think that also applies here for giving consent. I think I prefer a second `#newsletter` channel to bump visibility of the feed / process.

Since people will be giving consent in their messages, explicit consent in the messages would be much more visible than implicit consent based on a channel description, as people will see other people giving consent, and understanding what to do from that. So I don't think the same argument applies in this case.

The visibility argument you make is strong though.

**Reply by NiklasEi** [reply]
I think my idea of the process is very close to what you were suggesting/discussing here:

Posting in #showcase is an implicit opt-in to being pinged about your posts on Discord when the next newsletter is starting to be prepared. We can have a bot send messages like: "The next newsletter is starting to be prepared. You have posted content in #showcase [link to post(s)]. If you would like to publish something with the next newsletter, please take a look at [newsletter isssue PR] and submit your content until [deadline]."

Why?
* I agree with https://github.com/bevyengine/bevy/discussions/4560#discussioncomment-2618539 about the content in channel vs newsletter being different / for different audiences. I would not want my posts to automatically be included, but an automatic reminder would be very helpful.
* Automatic consent for a ping seems fine to me compared to automatic consent to publish my #showcase posts.
* This approach makes sure to point people to the PR, who might not be on GitHub (opposed to the "ping in PR" approach taken by the GameDev newsletter)
    * We could post these messages as threads in #newsletter so other people can help or completely take over content if wanted/consent is given. This might also make the messages more "legit" compared to private messages from a bot.

**Comment by HackerFoo** [comment]
The problem I have with the current policy is that, as a developer, I want to show off things that I'm excited about, but as a marketer(?), I want to control how my work is presented to a larger, possibly non-developer audience. I would like to present things in both places, but for the newsletter,  I'd like the opportunity to select and polish the content.

**Reply by wilk10** [reply]
Yes, i definitely share the same view. I would like to have a clear distinction in the way content is submitted for showcase vs the newsletter. 

I feel that future content for the newsletter from my side should be more polished than content for showcase, while I don't want to stress too much about what I post in showcase, nor that is going to be added to the newsletter "by mistake" (eg: I forget to remove consent, or I miss the ping from the bot after I posted)

**Comment by edgarssilva** [comment]
Maybe this could be worked on and be its own page on the website, where people can see what devs are working on each month.
If it eventually goes through a bot it could process it creating a page.

**Comment by nicopap** [comment]
I really like @NiklasEi 's suggestion of sending a "do you want your project to be featured on the newsletter?" (on discord, potentially on github too) to people who posted in #newsletter. This way, only a simple yes/no answer is required, and the assumed answer is "no". We could point to the newsletter PR too to give everyone complete control over their project presentation (like This Month in Rust Gamedev)
